src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -209,15 +216,54 @@ void remove_virtual_functionst::remove_virtual_function(
       t3->make_goto(t_final, true_exprt());
     }
 
+    // Emit target if end of dispatch table is reached or if the next element is
+    // dispatched to another function call. Assumes entries in the functions
+    // variable to be sorted for the identifier of the function to be called.
+    auto l_it = it;
+    l_it++;
+    bool next_emit_target = l_it == functions.crend();
+    if(!next_emit_target)
+    {
+      next_emit_target |=
```

Comment:
You know that `next_emit_target` is false, so `|=` is the same as `=` - seems fishy?